src/plugins/multifile/multifile.c

```diff
@@ -0,0 +1,693 @@
+		}
+		mc->getPhase = GETRESOLVER;
+	}
+	elektraPluginSetData (handle, mc);
+	if (rc == SUCCESS)
+		return 1;
+	else if (rc == NOUPDATE)
+		return 0;
+	else
+		return -1;
+}
+
+static Codes resolverSet (MultiConfig * mc, Key * parentKey)
+{
+	ksRewind (mc->childBackends);
+	Key * initialParent = keyDup (parentKey);
+	Key * k;
+	Codes rc = NOUPDATE;
+	while ((k = ksNext (mc->childBackends)) != NULL)
+	{
+		SingleConfig * s = *(SingleConfig **)keyValue (k);
+		if (s->rcResolver == NOUPDATE)
+		{
+			// fprintf (stderr, "SKIPPING %s:(%s)\n", s->parentString, s->fullPath);
+			continue;
+		}
+		else if (s->rcResolver == EMPTY)
+		{
+			// fprintf (stderr, "MARK FOR DELETE: %s:(%s)\n", s->parentString, s->fullPath);
+			++rc;
+			continue;
+		}
+		else
+		{
+			// fprintf (stderr, "UPDATE: %s:(%s)\n", s->parentString, s->fullPath);
+		}
+		
+		Plugin * resolver = s->resolver;
+		keySetName (parentKey, s->parentString);
+		keySetString (parentKey, s->fullPath);
+		int r = resolver->kdbSet (resolver, s->ks, parentKey);
+		s->rcResolver = rc = rvToRc (r);
+		if (rc == ERROR) break;
+		if (s->tmpFilename) elektraFree (s->tmpFilename);
+		s->tmpFilename = elektraStrDup (keyString (parentKey));
+		// fprintf (stderr, "tmp filename for %s: %s\n", s->fullPath, s->tmpFilename);
+	}
+	keySetName (parentKey, keyName (initialParent));
+	keySetString (parentKey, keyString (initialParent));
+	keyDel (initialParent);
+	return rc;
+}
+
+static Codes doSetStorage (MultiConfig * mc, Key * parentKey)
+{
+	ksRewind (mc->childBackends);
+	Key * initialParent = keyDup (parentKey);
+	Codes rc = NOUPDATE;
+	Key * k;
+	while ((k = ksNext (mc->childBackends)) != NULL)
+	{
+		SingleConfig * s = *(SingleConfig **)keyValue (k);
+		if (s->rcResolver == EMPTY)
+		{
+			rc = SUCCESS;
+			s->rcStorage = EMPTY;
+			continue;
+		}
+		else if (s->rcResolver != SUCCESS)
+		{
+			continue;
+		}
+		
+		keySetName (parentKey, s->parentString);
+		keySetString (parentKey, s->tmpFilename);
+		Plugin * storage = s->storage;
+		int r = storage->kdbSet (storage, s->ks, parentKey);
+		s->rcStorage = rc = rvToRc (r);
+
+		// fprintf (stderr, "%s ->kdbSet returned %d\n", mc->storage, r);
+
+		if (rc == ERROR) break;
+	}
+	keySetName (parentKey, keyName (initialParent));
+	keySetString (parentKey, keyString (initialParent));
+	keyDel (initialParent);
+	return rc;
+}
+
+static Codes doCommit (MultiConfig * mc, Key * parentKey)
+{
+	ksRewind (mc->childBackends);
+	Key * initialParent = keyDup (parentKey);
+	Codes rc = NOUPDATE;
+	Key * k;
+	while ((k = ksNext (mc->childBackends)) != NULL)
+	{
+		SingleConfig * s = *(SingleConfig **)keyValue (k);
+		if (s->rcStorage == EMPTY)
+		{
+			unlink (s->fullPath);
+			continue;
+		}
+		else if (s->rcStorage != 1)
+			continue;
+		
+		keySetName (parentKey, s->parentString);
+		keySetString (parentKey, s->fullPath);
+		Plugin * resolver = s->resolver;
+		int r = resolver->kdbSet (resolver, s->ks, parentKey);
+		rc = rvToRc (r);
+		if (rc == ERROR) break;
```

Comment:
i think in that case it would be clearer if we add this condition to the loop condition